api/api.go

```diff
@@ -0,0 +1,68 @@
+package api
+
+import (
+	"context"
+	"net/http"
+	"os"
+
+	"github.com/ONSdigital/dp-recipe-api/config"
+	"github.com/ONSdigital/dp-recipe-api/store"
+	"github.com/ONSdigital/go-ns/server"
+	"github.com/ONSdigital/log.go/log"
+	"github.com/gorilla/mux"
+)
+
+var srv *server.Server
+
+//RecipeAPI contains store and features for managing the recipe
+type RecipeAPI struct {
+	dataStore         store.DataStore
+	Router            *mux.Router
+	EnableMongoData   bool
+	EnableMongoImport bool
+}
+
+//CreateAndInitialiseRecipeAPI create a new RecipeAPI instance based on the configuration provided and starts the HTTP server.
+func CreateAndInitialiseRecipeAPI(ctx context.Context, cfg config.Configuration, dataStore store.DataStore) {
+	router := mux.NewRouter()
+	api := NewRecipeAPI(ctx, cfg, router, dataStore)
+
+	log.Event(ctx, "starting http server", log.INFO, log.Data{"bind_addr": cfg.BindAddr})
+	srv = server.New(cfg.BindAddr, api.Router)
+	if err := srv.ListenAndServe(); err != nil {
+		log.Event(ctx, "error starting http server for API", log.FATAL, log.Error(err))
+		os.Exit(1)
+	}
+}
+
+//NewRecipeAPI create a new Recipe API instance and register the API routes based on the application configuration.
+func NewRecipeAPI(ctx context.Context, cfg config.Configuration, router *mux.Router, dataStore store.DataStore) *RecipeAPI {
+	api := &RecipeAPI{
+		dataStore:         dataStore,
+		Router:            router,
+		EnableMongoData:   cfg.MongoConfig.EnableMongoData,
+		EnableMongoImport: cfg.MongoConfig.EnableMongoImport,
+	}
+
+	api.get("/health", api.HealthCheck)
+	api.get("/recipes", api.RecipeListHandler)
+	api.get("/recipes/{id}", api.RecipeHandler)
+	if api.EnableMongoImport {
+		api.Router.HandleFunc("/allrecipes", api.AddAllRecipeHandler).Methods("POST")
```

Comment:
I don't know if this belongs to another PR or this but I would suggest to keep the `/recipes` path here (instead of allrecipes) -

Reply:
The `/allrecipes` is an one-off endpoint. This endpoint is used to transfer all the existing recipes into mongo from the data.go file (where the recipes are all hard-coded). The migration to mongo has taken place, this endpoint will no longer be useful. 

Also, I am using `/recipes` endpoint with POST method to add recipe in mongo in another PR which i haven't send in `dev_code_review` as it is blocked because I need to do audit and authentication as it is dealing with POST request. Once the audit library has been fully updated, then i will complete the PR and push it for review

Reply:
ok thanks for the explanation!